crates/taker/src/routes.rs

```diff
@@ -305,20 +311,78 @@ pub async fn put_sync_wallet(taker: &State<Taker>, _user: User) -> Result<(), Ht
     Ok(())
 }
 
-#[rocket::get("/backup")]
-#[instrument(name = "GET /backup", skip_all)]
-pub async fn get_seed_backup(
-    seed: &State<Arc<ThreadSafeSeed>>,
+#[rocket::get("/export")]
+#[instrument(name = "GET /export", skip_all)]
+pub async fn get_export_seed<'a>(
+    data_dir: &'a State<PathBuf>,
+    environment: &'a State<Environment>,
     _user: User,
-) -> Result<DownloadResponsePro, HttpApiProblem> {
+) -> Result<DownloadResponsePro<'a>, HttpApiProblem> {
+    if environment.inner().as_string() == "umbrel" {
+        return Err(HttpApiProblem::new(StatusCode::BAD_REQUEST)
+            .title("Could not export seed file")
+            .detail("Cannot export seed file for umbrel environment"));
+    }
+
+    let seed = tokio::fs::read(data_dir.join(seed::TAKER_WALLET_SEED_FILE))
+        .await
+        .map_err(|e| {
+            HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
+                .title("Could not export seed file")
+                .detail(format!("{e:#}"))
+        })?;
+
     let resp = DownloadResponsePro::from_vec(
-        seed.inner().seed(),
+        seed,
         Some("taker_seed"),
         Some(mime::APPLICATION_OCTET_STREAM),
     );
     Ok(resp)
 }
 
+#[rocket::put("/import", data = "<seed>")]
+#[instrument(name = "PUT /import", skip_all)]
+pub async fn put_import_seed(
+    seed: Data<'_>,
+    taker: &State<Taker>,
+    data_dir: &State<PathBuf>,
+    network: &State<Network>,
+    environment: &State<Environment>,
+) -> Result<(), HttpApiProblem> {
+    if environment.inner().as_string() == "umbrel" {
+        return Err(HttpApiProblem::new(StatusCode::BAD_REQUEST)
+            .title("Could not import seed file")
+            .detail("Cannot import seed file for umbrel environment"));
+    }
+
+    // fetch seed from upload stream as bytes. only fetch max 256 bytes as the seed file shouldn't
+    // be bigger than that.
+    let seed = seed.open(256.bytes()).into_bytes().await.map_err(|e| {
```

Comment:
🔧 It would be nice if the `256` number came from a constant in the appropriate module. That way you can get rid of the comment too by adding documentation onto the constant.